stat/distuv/poisson.go

```diff
@@ -65,28 +65,55 @@ func (p Poisson) Prob(x float64) float64 {
 }
 
 // Rand returns a random sample drawn from the distribution.
+// NUMERICAL RECIPES IN C: THE ART OF SCIENTIFIC COMPUTING (ISBN 0-521-43108-5)
+// p. 294
+// <http://www.aip.de/groups/soe/local/numres/bookcpdf/c7-3.pdf>
 func (p Poisson) Rand() float64 {
-	// poisson generator based upon the multiplication of
-	// uniform random variates.
-	// see:
-	//  Non-Uniform Random Variate Generation,
-	//  Luc Devroye (p504)
-	//  http://www.eirene.de/Devroye.pdf
-	x := 0.0
-	prod := 1.0
-	exp := math.Exp(-p.Lambda)
+	g := math.Exp(-p.Lambda)
 	rnd := rand.Float64
 	if p.Source != nil {
 		rnd = p.Source.Float64
 	}
 
-	for {
-		prod *= rnd()
-		if prod <= exp {
-			return x
+	var sq, alxm float64
+	var em, t, y float64
+
+	var lg float64
+
+	if p.Lambda < 12.0 {
+		// Use direct method.
+		em = -1
```

Comment:
`var em int`

Reply:
Would you rather Rand() return type `int` or a `float64` that is an integer?

Reply:
`float64`

Reply:
If `em` is of type `int` it will need to go through a number of type conversions.

Reply:
So, `var em float64` is better, sorry.

Reply:
No worries. I was just asking for confirmation. I assumed I was likely overlooking something but wanted to confirm.